portal/static/js/sessionMonitor.js

```diff
@@ -1,48 +1,135 @@
-            .on("show.bs.modal", function() { sessMon.modalShown = true; })
-            .on("hide.bs.modal", function() { sessMon.modalShown = false; if (sessMon.intervalMonitor)  { clearInterval(sessMon.intervalMonitor); } })
-            .on("click", "#stay-logged-in", sessMon.extendsess)
-            .on("click", "#log-out", sessMon.logout);
-    var warningText = ($("#session-warning-modal").find("#remaining-time").text()).replace("{time}",(sessMon.timeBeforeWarning / 1000));
-    $("#session-warning-modal").find("#remaining-time").text(warningText);
-  };
-  this.readCookie = function(name) {
-    var nameEQ = name + "=";
-    var ca = document.cookie.split(";");
-    for (var i = 0; i < ca.length; i++) {
-        var c = ca[i];
-        while (c.charAt(0) === " ") {
-          c = c.substring(1, c.length);
-        }
-        if (c.indexOf(nameEQ) === 0) {
-          return c.substring(nameEQ.length, c.length);
-        }
-    }
-    return null;
+    var sessMon=(function(n, o) {
+      return function(t) {
+          "use strict";
+          var l = {},
+          g = new Date,
+          r, u, a = {
+                  sessionLifetime: 36e5,
+                  timeBeforeWarning: 6e5,
+                  minPingInterval: 6e4,
+                  activityEvents: "mouseup",
+                  pingUrl: n + "/api/ping",
+                  logoutUrl: n + "/logout",
+                  timeoutUrl: n + "/logout?timeout=1",
+                  ping: function() {
+                      $.ajax({
+                          type: "POST",
+                          contentType: "text/plain",
+                          headers: {
+                              "X-CSRFToken": o
+                          },
+                          cache: !1,
+                          url: l.pingUrl,
+                          crossDomain: !0
+                      });
+                  },
+                  logout: function() {
+                      window.location.href = l.logoutUrl;
+                  },
+                  onwarning: function() {
+                      var n = Math.round(l.timeBeforeWarning / 60 / 1e3),
+                          o = $('<div id="jqsm-warning">Your session will expire in ' + n + ' minutes. <button id="jqsm-stay-logged-in">Stay Logged In</button><button id="jqsm-log-out">Log Out</button></div>');
```

Comment:
Can you add translation tags? If it's an issue of not wanting to include the i18n JS libs, maybe you could keep the translation in a special element in the portal wrapper?

Reply:
@ivan-c ivan, that is the default configuration.  I have overridden that with the session modal element that *does* have translation tags around text content (i.e. in line 98 where the default on-warning config is overridden with the invocation of the aforementioned modal). The modal element is embedded in the portal wrapper's html.